piqueserver/core_commands/movement.py

```diff
@@ -17,49 +17,74 @@ def unstick(connection, player=None):
     player.set_location_safe(player.get_location())
 
 
-@command('goto', admin_only=True)
-def go_to(connection, value):
+@command('moves', admin_only = True)
+def move_silent(connection, *args):
     """
-    Go to a specified sector (e.g. A5) and inform everyone on the server of it
-    /goto <sector>
-    If you're invisivible, it will happen silenty
+    Silently move yourself or of a given player to a specified sector (e.g. A5).
+    If the x/y/z coordinate makes the player appear outside of the world bounds, take the bound instead
+    /moves <sector> [player]
+    /moves <x> <y> <z> [player]
     """
-    if connection not in connection.protocol.players:
-        raise KeyError()
-    move(connection, connection.name, value, silent=connection.invisible)
+    do_move(connection, args, True)
 
 
-@command('gotos', admin_only=True)
-def go_to_silent(connection, value):
+@command(admin_only=True)
+def move(connection, *args):
     """
-    Silently go to a specified sector
-    /gotos <sector>
+    Move yourself or of a given player to a specified sector (e.g. A5) and inform everyone on the server of it.
+    If you're invisivible, it will happen silenty.
+    If the x/y/z coordinate makes the player appear outside of the world bounds, take the bound instead
+    /move <sector> [player]
+    /move <x> <y> <z> [player]
     """
     if connection not in connection.protocol.players:
-        raise KeyError()
-    move(connection, connection.name, value, True)
+            raise ValueError()
+    do_move(connection, args)
+
+
+def do_move(connection, args, silent = False):
+    position = None
+    player = None
+    nbArgs = len(args)
+
+    #case position is <sector>
+    if nbArgs == 1 or nbArgs == 2:
+        x, y = coordinates(args[0])
+        x += 32
+        y += 32
+        z = connection.protocol.map.get_height(x, y) - 2
+        position = args[0].upper()
+    #case position is <x> <y> <z>
+    elif nbArgs == 3 or nbArgs == 4:
+        x = min(max(0, int(args[0])), 511)
+        y = min(max(0, int(args[1])), 511)
+        z = min(max(0, int(args[2])), connection.protocol.map.get_height(x, y) - 2)
+        position = '%d %d %d' % (x, y, z)
+    else:
+        return 'Wrong number of parameters!'
```

Comment:
It would be better to raise an error here, as this automatically displays the command usage in-game.

Reply:
Should I raise the exception with this message instead or just raise ValueError?

Reply:
Either is fine